test/topology_custom/test_raft_recovery_entry_loss.py

```diff
@@ -0,0 +1,152 @@
+#
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import logging
+import time
+import pytest
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.util import check_system_topology_and_cdc_generations_v3_consistency, \
+        check_token_ring_and_group0_consistency, delete_discovery_state_and_group0_id, delete_raft_group_data, \
+        reconnect_driver, wait_for_cdc_generations_publishing
+from test.topology_custom.test_group0_schema_versioning import get_group0_schema_version, get_local_schema_version
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_raft_recovery_entry_lose(manager: ManagerClient):
+    """
+    Test that the new Raft recovery procedure works corrrectly if some committed group 0 entry has been permanently lost
+    (it has been committed only by dead nodes).
+
+    1. Start a cluster with 5 nodes
+    2. Create a scenario where nodes have the following group 0 states:
+    - node 1: v1,
+    - node 2: v2,
+    - nodes 3-5: v3.
+    3. Kill nodes 3-5 causing a permanent group 0 majority loss. We also permanently lose v3. After recovering majority,
+    nodes 1-2 shoud have v2, which is the only safe option.
+    4. Run the recovery procedure to recreate group 0 with nodes 1, 2 as the only members. We make sure node 2
+    becomes the leader of new group 0 since it has a newer group 0 state. When node 1 joins new group 0, it receives
+    group 0 snapshot with v2 from node 2 and applies it. If node 1 became the leader, both nodes would end up with v1.
```

Comment:
If 1 became the leader, 2 would remain with v2 due to mutation timestamps. The snapshot transfer from 1 wouldn't do anything. Probably. Recall that snapshot transfer does not purge group 0 tables on destination before applying the mutation. It merges the mutation from snapshot with existing on-disk one.

Reply:
Right. This makes using the wrong recovery leader even more dangerous. Moving all nodes to the same older state could be unsafe only in a few specific cases (mentioned in the "Recovery of tablet metadata after group0 majority loss" doc), but moving nodes to different states basically brakes Raft invariants.